### PR TITLE
tests(navs): Add `PageNavbar` and `NavsetHidden` tests and controllers

### PR DESCRIPTION
**Pull request by karangattu**
Add newer controllers and refactor some code to leverage mixin to reduce duplication.
Also, add tests for navset_hidden and page_navbar.

### CODE REVIEW

File: tests/playwright/shiny/components/nav/navset_hidden_kitchensink/test_navset_hidden_kitchensink.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+    navset_hidden_1 = controller.NavsetHidden(page, "hidden_tabs1")
+    radio1 = controller.InputRadioButtons(page, "controller1")
+    navset_hidden_1.expect_value("panel2")
+    navset_hidden_1._expect_content_text("Panel 2 content")
+    # assert the DOM structure for hidden_navset with header and footer is preserved
+    assert (
+        page.locator(
+            "#hidden_tabs1 + #navset_hidden_header1 + .tab-content + #navset_hidden_footer1"
+        ).count()
```
Comment (karangattu) [inline tests/playwright/shiny/components/nav/navset_hidden_kitchensink/test_navset_hidden_kitchensink.py:18]: use playwright_expect().to_have_count(1) instead

File: shiny/playwright/controller/_navs.py [hunk 2]
```diff
@@ -631,20 +638,30 @@ def expect_position(
                 re.compile(rf"{position}"), timeout=timeout
             )
 
-    def expect_inverse(self, *, timeout: Timeout = None) -> None:
+    def expect_inverse(
+        self,
+        value: bool,
```
Comment (schloerke) [inline shiny/playwright/controller/_navs.py:643]: This new parameter will need a changelog entry

Also:
* `.expect_layout()` -> `.expect_fluid()`
* Added `PageNavbar` class

File: shiny/playwright/controller/_navs.py [hunk 3]
```diff
@@ -424,18 +429,20 @@ def __init__(self, page: Page, id: str) -> None:
...
-            `True` if the navset pill list is expected to have a well, `False` otherwise.
+        value
+            `True` if the navset pill list is expected to be constructed with a well,
+            `False` otherwise.
         timeout
             The maximum time to wait for the expectation to pass. Defaults to `None`.
         """
-        if has_well:
+        if value:
             playwright_expect(self.loc_container.locator("..")).to_have_class("well")
```
Comment (karangattu) [inline shiny/playwright/controller/_navs.py:446]: TODO-karan: Change to `expect_to_have_class(loc, has_class=value)`

File: shiny/playwright/controller/_navs.py [hunk 4]
```diff
@@ -670,28 +690,117 @@ def expect_gap(self, gap: str, *, timeout: Timeout = None) -> None:
...
+        timeout
+            The maximum time to wait for the expectation to pass. Defaults to `None`.
+        """
+        # This is important since fillable_mobile needs fillable property to be True
+        expect_to_have_class(
+            self.page.locator("body"), "bslib-page-fill", timeout=timeout
+        )
+
+        if value:
+            expect_not_to_have_class(
```
Comment (karangattu) [inline shiny/playwright/controller/_navs.py:767]: switch to `_expect_to_have_class`

**Pull request merged**